src/radical/pilot/agent/agent_0.py

```diff
@@ -194,9 +188,25 @@ def initialize(self):
         self.rpc('prepare_env', env_name='rp', env_spec=env_spec,
                                 rpc_addr=self._pid)
 
-        # start any services if they are requested
+        # ...and prepare all envs which are defined in the pilot description
+        for env_name, env_spec in self._cfg.get('prepare_env', {}).items():
```

Comment:
`self._cfg.get(` -> `self.session.cfg.get(` (or even `self.session.cfg.prepare_env`) to use one source for config data

Reply:
I would consider the reverse: `self._cfg` should be normative for components.  Pushed that now...